compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpServiceValidator.java

```diff
@@ -265,7 +282,29 @@ private static void validateServiceConfigAnnotation(SyntaxNodeAnalysisContext ct
                         break;
                     }
                 }
+                if (INTERCEPTORS_ANNOTATION_FIELD.equals(strings[0].trim())) {
+                    interceptorsFieldLocation = Optional.of(field.location());
+                }
+            }
+        }
+        return interceptorsFieldLocation;
+    }
+
+    private static void validateServiceInterceptorDefinitions(SyntaxNodeAnalysisContext ctx,
+                                                              Optional<NodeLocation> interceptableSvcLocation,
+                                                              Optional<NodeLocation> interceptorsFieldLocation) {
+        if (interceptorsFieldLocation.isPresent()) {
+            if (interceptableSvcLocation.isPresent()) {
+                DiagnosticInfo diagnosticInfo = new DiagnosticInfo(HTTP_153.getCode(),
+                        HTTP_153.getMessage(), HTTP_153.getSeverity());
+                ctx.reportDiagnostic(DiagnosticFactory
+                        .createDiagnostic(diagnosticInfo, interceptableSvcLocation.get()));
+                ctx.reportDiagnostic(DiagnosticFactory
+                        .createDiagnostic(diagnosticInfo, interceptorsFieldLocation.get()));
             }
+            DiagnosticInfo diagnosticInfo = new DiagnosticInfo(HTTP_201.getCode(),
```

Comment:
Can we skip giving a warning if there is already an error?

Reply:
Will keep both warning and error as discussed